Documentation/coding-guidelines/breaking-change-rules.md

```diff
@@ -163,6 +163,12 @@ Breaking Change Rules
 
 * Changing a `struct` type to a `ref struct` type and vice versa
 
+* Changing the underlying type of an enum to a different size
+
+    This is a binary breaking change in a category all its own: not just at runtime but also while passively parsing the assembly’s attributes.
+
+    The ECMA 335 format requires the reader to resolve the enum’s assembly and look up the enum’s underlying data size, given the enum’s serialized type name, in order to know how many bytes to read as the attribute value. So if a value of the enum has been used as an attribute argument, and the enum data size in the resolved assembly differs from the enum data size when the assembly was compiled, the assembly reader will have no option but to read too much or too little and misread the current attribute argument and all following arguments. (The reader may not necessarily realize it is misinterpreting the data unless the resolved enum data size is smaller, which causes the reader to read past the end of the available data.)
```

Comment:
There are many more main stream cases that this breaks. I do not think that highlighting the custom attribute case here helps clarity.